File: kotlin/isbn-verifier/src/main/kotlin/IsbnVerifier.kt
class IsbnVerifier {
  fun isValid(s: String): Boolean {
    var sum = 0
    val stripped = Regex("""[^0-9X]""").replace(s.toUpperCase(), "")
    if (stripped.length < 10) {
      return false
    }
    for (i in 0..stripped.lastIndex) {
      if (stripped[i] == 'X') {
        if (i != stripped.lastIndex) {
          return false
        }
        sum += 10
      } else {
        sum += (stripped[i].toInt() - '0'.toInt()) * (10 - i)
      }
    }
    return sum % 11 == 0
  }
}

File: dart/isbn-verifier/lib/isbn_verifier.dart
bool isValid(String input) {
  var stripped = input.toUpperCase().replaceAll(new RegExp(r'[^0-9X]'), '');
  if (stripped.length != 10) {
    return false;
  }
  var sum = 0;
  for(int i = 0; i < stripped.length; i++) {
    if (stripped[i] == 'X') {
        if (i == 9) {
          sum += 10;
        } else {
          return false;
        }
    } else {
      sum += int.parse(stripped[i]) * (10 - i);
    }
  }
  return sum % 11 == 0;
}